test/testground/network/consensus_node.go

```diff
@@ -0,0 +1,326 @@
+	if err != nil {
+		return nil, err
+	}
+
+	packets, err := DownloadSync(ctx, initCtx, PeerPacketTopic, PeerPacket{}, runenv.TestInstanceCount)
+	if err != nil {
+		return nil, err
+	}
+
+	return packets, nil
+}
+
+// Init creates the files required by tendermint and celestia-app using the data
+// downloaded from the Leader node.
+func (cn *ConsensusNode) Init(baseDir string, genesis json.RawMessage, mcfg RoleConfig) error {
+	cn.CmtConfig = mcfg.CmtConfig
+	cn.AppConfig = mcfg.AppConfig
+	cn.AppCreator = cmd.NewAppServer
+	cn.SuppressLogs = true
+
+	// manually set the protocol version to the one used by the testground
+	appOpts := testnode.DefaultAppOptions()
+	appOpts.Set(app.SetProtocolVersionOptionKey, tg.Version)
+	cn.AppOptions = appOpts
+
+	baseDir = filepath.Join(baseDir, ".celestia-app")
+	cn.baseDir = baseDir
+
+	cn.CmtConfig.SetRoot(baseDir)
+
+	// save the genesis file
+	configPath := filepath.Join(baseDir, "config")
+	err := os.MkdirAll(configPath, os.ModePerm)
+	if err != nil {
+		return err
+	}
+	// save the genesis file as configured
+	err = cmtos.WriteFile(cn.CmtConfig.GenesisFile(), genesis, 0o644)
+	if err != nil {
+		return err
+	}
+	pvStateFile := cn.CmtConfig.PrivValidatorStateFile()
+	if err := tmos.EnsureDir(filepath.Dir(pvStateFile), 0o777); err != nil {
+		return err
+	}
+	pvKeyFile := cn.CmtConfig.PrivValidatorKeyFile()
+	if err := tmos.EnsureDir(filepath.Dir(pvKeyFile), 0o777); err != nil {
+		return err
+	}
+	filePV := privval.NewFilePV(cn.consensusKey, pvKeyFile, pvStateFile)
+	filePV.Save()
+
+	nodeKeyFile := cn.CmtConfig.NodeKeyFile()
+	if err := tmos.EnsureDir(filepath.Dir(nodeKeyFile), 0o777); err != nil {
+		return err
+	}
+	nodeKey := &p2p.NodeKey{
+		PrivKey: cn.networkKey,
+	}
+	if err := nodeKey.SaveAs(nodeKeyFile); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// StartNode uses the testnode package to start a tendermint node with
+// celestia-app and the provided configuration.
+func (cn *ConsensusNode) StartNode(ctx context.Context, baseDir string) error {
+	ucfg := cn.UniversalTestingConfig()
+	tmNode, app, err := testnode.NewCometNode(baseDir, &ucfg)
+	if err != nil {
+		return err
+	}
+
+	cn.cmtNode = tmNode
+	cctx := testnode.NewContext(ctx, cn.kr, ucfg.TmConfig, cn.params.ChainID)
+
+	cctx, stopNode, err := testnode.StartNode(tmNode, cctx)
+	cn.stopFuncs = append(cn.stopFuncs, stopNode)
+	if err != nil {
+		return err
+	}
+
+	cctx, cleanupGRPC, err := testnode.StartGRPCServer(app, ucfg.AppConfig, cctx)
+	cn.stopFuncs = append(cn.stopFuncs, cleanupGRPC)
+
+	cn.cctx = cctx
+
+	return err
+}
+
+// Stop stops the node and cleans up the data directory by calling the cleanup
+// functions. It returns the last error that was not nil if any of the cleanup
+// functions returned an error.
+func (cn *ConsensusNode) Stop() error {
+	var err error
+	for _, stop := range cn.stopFuncs {
+		if sterr := stop(); sterr != nil {
+			err = sterr
+		}
+	}
+	return err
```

Comment:
cool idea!

Reply:
I originally left this here as this would only have an effect on a single run. Calling the stop functions is almost pointless. We can adopt the usage of multierror tho for good practice